lib/Transforms/YosysOptimizer/YosysOptimizer.cpp

```diff
@@ -0,0 +1,122 @@
+#include "include/Transforms/YosysOptimizer/YosysOptimizer.h"
+
+#include "include/Dialect/Comb/IR/CombOps.h"
+#include "include/Target/Verilog/VerilogEmitter.h"
+#include "kernel/yosys.h"  // from @at_clifford_yosys
+#include "lib/Transforms/YosysOptimizer/LUTImporter.h"
+#include "llvm/include/llvm/Support/FormatVariadic.h"   // from @llvm-project
+#include "mlir/include/mlir/Dialect/Arith/IR/Arith.h"   // from @llvm-project
+#include "mlir/include/mlir/Dialect/Func/IR/FuncOps.h"  // from @llvm-project
+#include "mlir/include/mlir/IR/IRMapping.h"             // from @llvm-project
+#include "mlir/include/mlir/Pass/PassManager.h"         // from @llvm-project
+#include "mlir/include/mlir/Transforms/Passes.h"        // from @llvm-project
+
+namespace mlir {
+namespace heir {
+
+// $0: verilog filename
+// $1: function name
+// $2: yosys runfiles
+// $3: abc path
+// $4: abc fast option -fast
+constexpr std::string_view kYosysTemplate = R"(
+read_verilog {0};
+hierarchy -check -top \{1};
+proc; memory;
+techmap -map {2}/techmap.v; opt;
+abc -exe {3} -lut 3 {4};
+opt_clean -purge;
+rename -hide */c:*; rename -enumerate */c:*;
+techmap -map {2}/map_lut_to_lut3.v; opt_clean -purge;
+hierarchy -generate * o:Y i:*; opt; opt_clean -purge;
+clean;
+)";
+
+struct YosysOptimizer
+    : public PassWrapper<YosysOptimizer, mlir::OperationPass<mlir::ModuleOp>> {
```

Comment:
Actually, just put it in a github issue and assign it to me. I think I can make it work.

Reply:
Oh yes! The initial reason was because of the custom init method which eventually left over the defunct tablegen - I think the issue was I also have custom registration methods, so would probably need to remove the directives to generate the pass reg methods that conflicted with what I wanted. If it's not as trivial as that, I'll create a GH issue!